komodo/yaml_file_type.py

```diff
@@ -38,16 +134,506 @@ class ManifestFile(YamlFile):
     Return the data from 'manifest' YAML, but validate it first.
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.yml = None
+
     def __call__(self, value: str) -> Dict[str, Dict[str, str]]:
-        yaml = super().__call__(value)
+        yml = super().__call__(value)
+        self.validate_manifest_file(yml)
+        return yml
+
+    @staticmethod
+    def validate_manifest_file(manifest_file_content: dict):
         message = (
             "The file you provided does not appear to be a manifest file "
             "produced by komodo. It may be a release file. Manifest files "
             "have a format like the following:\n\n"
             "python:\n  maintainer: foo@example.com\n  version: 3-builtin\n"
             "treelib:\n  maintainer: foo@example.com\n  version: 1.6.1\n"
         )
-        for _, metadata in yaml.items():
-            assert isinstance(metadata, dict), message
-            assert isinstance(metadata["version"], str), message
-        return yaml
+        assert isinstance(manifest_file_content, dict), message
+        errors = []
+        for package_name, metadata in manifest_file_content.items():
+            if not isinstance(metadata, dict):
+                errors.append(f"Invalid metadata for package '{package_name}'")
+                continue
+            if not isinstance(metadata["version"], str):
+                errors.append(
+                    f"Invalid version type in metadata for package '{package_name}'"
+                )
+        handle_validation_errors(errors, message)
+
+
+class RepositoryFile(YamlFile):
+    """
+    Return the data from 'repository' YAML, but validate it first.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.yml = None
```

Comment:
Typing for this class variable would be nice!

Reply:
I also think that `yml` is a bad variable name, because it is not yaml any longer. Further down you call it `repository_entries`, that might be better.

Reply:
I changed the key to content, as it refers to the content of the file. Should I call it content_dict instead to better indicate the type? This should be shown in most IDEs as I have added typing, but it could also be more clear.